### ISSUE TITLE
Difference of behaviour between serve and http-server

### ISSUE THREAD

**Issue by eric-burel** [vercel/serve#631]
Hi, we have noticed some discrepencies between `serve` and `http-server` when opening a Storybook static export.
Issue is described here: https://github.com/storybookjs/storybook/issues/10933

To sum it up, with `http-server` the first story opens correctly, but not with `serve`, which seems a bit less reliable for yet unknown reasons.
![image](https://user-images.githubusercontent.com/7983005/92627553-cecc4d80-f2cb-11ea-8f95-eecb085d2b16.png)

While you should get this:
![image](https://user-images.githubusercontent.com/7983005/92627606-e4da0e00-f2cb-11ea-805c-63308cbe9908.png)

I have a reproduction, but not minimal, in Vulcan Next. That may still help to debug.

```
git clone https://github.com/VulcanJS/vulcan-next.git
cd vulcan-next
yarn
yarn build:storybook
serve storybook-static # this will mostly work but not reliably, while http-server will work ok
```

**Comment by theKashey** [comment]
serve needs at least `cleanUrl:false` option

**Comment by eric-burel** [comment]
`cleanUrls` seems to break everything for me:
![image](https://user-images.githubusercontent.com/7983005/98080946-946bd080-1e76-11eb-80ab-96d6fe8ad6db.png)

**Comment by eric-burel** [comment]
See https://github.com/vercel/next.js/pull/18775 for a reproducible minimal example

**Comment by Heziode** [comment]
I have (I think) the same issue with [my project](https://github.com/FOE-Tools/FOE-Tools.github.io).

If I build the project `yarn run generate:gh` and `serve dist`, the website doesn't work well, and many pages/data are unavailable. I do not have this issue with `http-serve dist`.

It's annoying, because I can't deploy my site on Vercel (which has exactly the same problem).

Having to wait a fix before moving from GHPage to Vercel… 😭

**Comment by eric-burel** [comment]
Haven't checked in a while! Do you still have this issue? You'll probably want to access the right URL directly, maybe smth like "/index.html".

**Comment by baptisteArno** [comment]
Yep, problem still exists! We ended up using [static-web-server](https://github.com/joseluisq/static-web-server).

**Comment by Wxh16144** [comment]
I'm curious what the difference is between `serve` and `https-server`🤔

**Comment by bchewy** [comment]
This is still an issue. We've tried with `serve.json` as config
```json
"cleanUrls": true,
    "trailingSlash": true,
    "rewrites": [
        {
            "source": "/iframe.html",
            "destination": "/iframe.html"
        },
        ...
```
